src/coord.jl

```diff
@@ -24,13 +24,19 @@ Coord(val::T=:) where T = Coord{T,AutoMode,NoMetadata}(val, AutoMode(), NoMetada
 dims(d::Coord) = dims(mode(d))
 bounds(d::Coord) = ntuple(i -> extrema((x[i] for x in val(d))), length(first(d)))
 
-# Return a Vector{Bool} for matching coordinates  
+# Return a Vector{Bool} for matching coordinates
 sel2indices(dim::Coord, sel::DimTuple) = sel2indices(dim, sortdims(sel, dims(dim)))
-sel2indices(dim::Coord, sel::Tuple) = [all(map(_matches, sel, x)) for x in val(dim)] 
+sel2indices(dim::Coord, sel::Tuple) = [all(map(_matches, sel, x)) for x in val(dim)]
 sel2indices(dim::Coord, sel::StandardIndices) = sel
 
-_matches(sel::Dimension, x) = _matches(val(sel), x) 
+_matches(sel::Dimension, x) = _matches(val(sel), x)
 _matches(sel::Between, x) = (x >= first(sel)) & (x < last(sel))
 _matches(sel::At, x) = x == val(sel)
 _matches(sel::Colon, x) = true
 _matches(sel::Nothing, x) = true
+
+# Fix for https://github.com/rafaqz/DimensionalData.jl/issues/263
+_tozerovector(x) = zero(x)
```

Comment:
`_tozero` is more accurate, and just always using map catches more types - e.g. regular `Vector` has no `zero`.

```suggestion
_tozero(xs) = map(x -> zero(x)), xs)
```